src/graph.rs

```diff
@@ -201,10 +199,10 @@ impl<'tree> Graph<'tree> {
                     Value::String(string) => serializer.serialize_str(string),
                     // FIXME: there's no way to distinguish sets and lists, so we can't roundtrip accurately
                     Value::List(list) => {
-                        serializer.collect_seq(list.iter().map(|value| JSONValue(value, ctx)))
+                        serializer.collect_seq(list.iter().map(|value| InContext(value, ctx)))
                     }
                     Value::Set(set) => {
-                        serializer.collect_seq(set.iter().map(|value| JSONValue(value, ctx)))
+                        serializer.collect_seq(set.iter().map(|value| InContext(value, ctx)))
                     }
                     // FIXME: we don't distinguish between syntax tree node IDs, graph node IDs, and integers
```

Comment:
Do syntax node IDs have much meaning in the output? Not that I can think of a better thing to output here...

Reply:
I can't imagine so, unless you have the syntax tree as well (which you don't). I was thinking about this, and maybe we could get a rough summary of the node: its rule name, its range, etc. Again, I'd like to punt on that for just a little while if possible.

Reply:
:+1: to punt.  I do think it's worth including the syntax tree (possibly with an option to give you (a) the whole tree, or (b) just the nodes mentioned in the output graph), but that doesn't have to block this PR.

Reply:
Great idea 👍🏻